rust/agents/relayer/src/msg/processor.rs

```diff
@@ -63,92 +66,106 @@ impl MessageProcessor {
+
+            // Skip if not whitelisted.
+            if !self.whitelist.msg_matches(&message, true) {
+                debug!(
+                    id=?message.id(),
+                    destination=message.destination,
+                    nonce=message.nonce,
+                    whitelist=?self.whitelist,
+                    "Message not whitelisted, skipping");
+                self.message_nonce += 1;
+                return Ok(());
+            }
+
+            // Skip if the message is blacklisted
+            if self.blacklist.msg_matches(&message, false) {
+                debug!(
+                    id=?message.id(),
+                    destination=message.destination,
+                    nonce=message.nonce,
+                    blacklist=?self.blacklist,
+                    "Message blacklisted, skipping");
+                self.message_nonce += 1;
+                return Ok(());
+            }
+
+            // Skip if the message is intended for a destination we do not service
+            if self.send_channels.get(&message.destination).is_none() {
+                debug!(
+                    id=?message.id(),
+                    destination=message.destination,
+                    nonce=message.nonce,
+                    "Message destined for unknown domain, skipping");
+                self.message_nonce += 1;
+                return Ok(());
+            }
+
+            // Feed the message to the prover sync
+            self.prover_sync
+                .write()
+                .await
+                .update_to_index(message.nonce)
+                .await?;
 
-        // Skip if the message is blacklisted
-        if self.blacklist.msg_matches(&message, false) {
             debug!(
-                id=?message.id(),
-                destination=message.destination,
-                nonce=message.nonce,
-                blacklist=?self.blacklist,
-                "Message blacklisted, skipping");
-            self.message_nonce += 1;
-            return Ok(());
-        }
+                msg_id=?message.id(),
+                msg_nonce=message.nonce,
+                "Sending message to submitter"
+            );
 
-        // Feed the message to the prover sync
-        self.prover_sync
-            .write()
-            .await
-            .update_to_index(message.nonce)
-            .await?;
-
-        debug!(
-            msg_id=?message.id(),
-            msg_nonce=message.nonce,
-            "Sending message to submitter"
-        );
-        // Finally, build the submit arg and dispatch it to the submitter.
-        let submit_args = SubmitMessageArgs::new(message.clone());
-        if let Some(send_channel) = self.send_channels.get(&message.destination) {
+            // Finally, build the submit arg and dispatch it to the submitter.
+            let submit_args = SubmitMessageArgs::new(message.clone());
+            let send_channel = self.send_channels.get(&message.destination).unwrap();
```

Comment:
Will this `unwrap` create a regression? Before it would just debug print if it did not exist.

Reply:
It should be the same since we check if it's none on line 138, so here it's already guaranteed to exist. I'll add a comment.